### PR TITLE
feat: update nginx and docker-compose files to support HTTPS.

### PR DESCRIPTION
**Pull request by Kennytian**
# Description

Please include a summary of the changes and which issues are fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement，including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

I have tested the code I submitted locally and everything works fine. Additionally, this change supports both HTTP and HTTPS requests. To enable HTTPS support, simply uncomment the code in the `docker/nginx/conf.d/default.conf` file and put the SSL certificate files in the `docker/nginx/ssl/` directory.

# Suggested Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests passed locally with my changes

### CODE REVIEW

**Comment by Kennytian** [comment]
@takatost  Please review my PR :)

File: docker/docker-compose.yaml [hunk 1]
```diff
@@ -326,8 +326,10 @@ services:
       - ./nginx/nginx.conf:/etc/nginx/nginx.conf
       - ./nginx/proxy.conf:/etc/nginx/proxy.conf
       - ./nginx/conf.d:/etc/nginx/conf.d
+      - ./nginx/ssl:/etc/ssl
```
Comment (crazywoola) [inline docker/docker-compose.yaml]:
```suggestion
      #- ./nginx/ssl:/etc/ssl
```

File: docker/docker-compose.yaml [hunk 2]
```diff
@@ -326,8 +326,10 @@ services:
...
     depends_on:
       - api
       - web
     ports:
       - "80:80"
+      - "443:443"
```
Comment (crazywoola) [inline docker/docker-compose.yaml]:
```suggestion
      #- "443:443"
```

**Review by crazywoola — COMMENTED** [review]
I'd recommend commenting on these lines, so that you can remove them when you need to, rather than having port 443 open by default in the first place

**Comment by Kennytian** [comment]
@crazywoola Thanks for your reply, I agree with you, and then will commenting on these lines right now.

**Review by crazywoola — APPROVED** [review]

**Pull request merged**